### DISCUSSION TITLE
Automatic static optimization appears not to be working?

### DISCUSSION THREAD

**Discussion by robbienohra** [vercel/next.js#21907] [Help] [answered]
The following link [Automatic Static Optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization) suggests that in the absence of `getServerSideProps` and `getInitialProps` that HTML files will be generated as per

```
.next/server/static/${BUILD_ID}/about.html
```

However if you even just pull a hello world example such as [Hello World](https://github.com/vercel/next.js/tree/canary/examples/hello-world) with simple pages such 
```
export default function DayPage() {
  return <div>Hello Day</div>
}
```

and run `yarn build` there are no HTML files to be found in the `.next/server/static` directory in fact there is no `.next/server/static` directory

HTML files are however seen in the `server/pages` directory

Am I missing something?

**Comment by michrome** [comment]
I think you’re looking for [`next export`](https://nextjs.org/docs/advanced-features/static-html-export).

**Reply by robbienohra** [reply]
no I'm not lol

nowhere in the above link does it state that Automatic Static Optimization is contingent on using `next export`

> This feature allows Next.js to emit hybrid applications that contain both server-rendered and statically generated pages.

**Reply by robbienohra** [reply]
In fact it states explicitly here [Advanced Features: Static HTML Export | Next.js](https://nextjs.org/docs/advanced-features/static-html-export)

>If you're looking to make a hybrid site where only some pages are prerendered to static HTML, Next.js already does that automatically for you! Read up on Automatic Static Optimization for details.

**Reply by robbienohra** [reply]
hence my confusion 😃

**Reply by robbienohra** [reply]
I admittedly may be misreading the docs — but the way it's phrased makes it sound like `next export` is when you're looking to serve the app without the use of Node at all analogous to Gatsby.

If `next export` is required to activate Automatic Static Optimization the doc seems to be suggesting the opposite 😋

**Reply by robbienohra** [reply]
you get what I mean?

Like the literal claim is:

> `next build` will emit .html files for statically optimized pages.

and then they specify where

`.next/server/static/${BUILD_ID}/about.html`

**Reply by robbienohra** [reply]
I do see HTML generated in `.next/server/pages` so may boil down to a typo in the docs? 

anyways hopefully we'll get clarification!

**Reply by michrome** [reply]
Ah, gotcha! I see below it was just a typo. 🙃

**Comment by ijjk** [comment, marked as answer]
Hi, it is expected that the server HTML and JS outputs are located in the `.next/server/pages` folder, the folder that was used has changed and the documentation does need to updated as well. I opened a PR [here](https://github.com/vercel/next.js/pull/21910) to update this

**Reply by robbienohra** [reply]
Excellent, thank you!

**Reply by robbienohra** [reply]
@ijjk quick question for ya

would this have any impact on an NGINX configuration?

I'm currently using the following routes for build assets 

```
location /_next/static
```

```
location /static
```

**Reply by ijjk** [reply]
No it shouldn't have any impact on NGINX configuration if you are using `next start` with `proxy_pass`
